pywhy_stats/api.py

```diff
@@ -59,32 +60,37 @@ def independence_test(
 
     See Also
     --------
-    fisherz : Fisher's Z test for independence
-    kci : Kernel Conditional Independence test
+    pywhy_stats.independence.fisherz : Fisher's Z test for independence
+    pywhy_stats.independence.kci : Kernel Conditional Independence test
     """
     method_module: ModuleType
     if method == Methods.AUTO:
         method_module = Methods.KCI
+    elif not isinstance(method, Methods):
+        raise ValueError(
+            f"Invalid method type. Expected one of {Methods.__members__.keys()}, "
+            f"but got {method}."
+        )
     else:
-        method_module = method
+        method_module = method  # type: ignore
 
     if method_module == Methods.FISHERZ:
         if condition_on is None:
             data = [X, Y]
         else:
             data = [X, Y, condition_on]
         for _data in data:
-            _, pval = scipy.stats.normaltest(_data)
+            res = scipy.stats.normaltest(_data, axis=0)
 
             # XXX: we should add pinguoin as an optional dependency for doing multi-comp stuff
-            if pval < 0.05:
+            if np.atleast_1d(res.pvalue).any() < 0.05:
                 warn(
                     "The provided data does not seem to be Gaussian, but the Fisher-Z test "
                     "assumes that the data follows a Gaussian distribution. The result should "
                     "be interpreted carefully or consider a different independence test method."
                 )
 
     if condition_on is None:
-        return method_module.ind(X, Y, method, **kwargs)
+        return method_module.value.ind(X, Y, **kwargs)
```

Comment:
The `value` part is very confusing, why do we need this? Or, why don't we keep the structure as before?

Reply:
Because it actually did not previously work. Since there was no test, we didn't catch it. 

Take `method_module = Enum(FISHERZ)` for example. 

`method_module.name == 'FISHERZ'` and `method_module.value == fisherz` (the function). Enumeration is kinda wonky imo because of this, but it does make things like `method_module == METHODS.FISHERZ` easier

Reply:
Oh I see. Can you add a test then to make sure that all methods in this high-level function is working with the enums?

Reply:
Added in `tests/test_api.py`